app/src/main/java/ch/epfl/sdp/drone3d/service/storage/dao/FirebaseMappingMissionDao.kt

```diff
@@ -128,10 +140,76 @@ class FirebaseMappingMissionDao @Inject constructor(
         return getMappingMissions(null)
     }
 
+    override fun getPrivateFilteredMappingMissions(): LiveData<List<MappingMission>> {
+        return privateFilteredMappingMissions
+    }
+
+    override fun getSharedFilteredMappingMissions(): LiveData<List<MappingMission>> {
+        return sharedFilteredMappingMissions
+    }
+
+    /**
+     * Update the list of mapping missions filtered by the new [filter]
+     * If the [ownerUid] is not null the missions are fetch from private repo of the owner
+     * Otherwise in the shared repo
+     */
+    private fun updateFilteredMappingMissions(ownerUid: String?, filter: String) {
+        val rootRef =
+            if (ownerUid != null) privateMappingMissionRef(ownerUid) else sharedMappingMissionRef()
+        val mappingMissions =
+            if (ownerUid != null) privateFilteredMappingMissions else sharedFilteredMappingMissions
+        val query =
+            if (ownerUid != null) privateFilteredMappingMissionsQuery else sharedFilteredMappingMissionsQuery
+        val listener =
+            if (ownerUid != null) privateFilteredMappingMissionsListener else sharedFilteredMappingMissionsListener
+
+        if (query != null && listener != null) {
+            query.removeEventListener(listener)
+        }
+
+        val missionsQuery = rootRef.orderByChild("nameUpperCase").startAt(filter).endAt(filter + "\uf8ff")
+        val missionsListener = missionsQuery.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+                val missionsSnapshot = dataSnapshot.children.map { c ->
+                    c.getValue(MappingMission::class.java)!!
+                }
+                mappingMissions.value = missionsSnapshot
+            }
+
+            override fun onCancelled(databaseError: DatabaseError) {
+                Timber.tag(TAG)
+                    .w(databaseError.toException(), "getFilteredMappingMissions:onCancelled")
+            }
+        })
+
+        if (ownerUid != null) {
+            privateFilteredMappingMissionsFilter = filter
+            privateFilteredMappingMissionsQuery = missionsQuery
+            privateFilteredMappingMissionsListener = missionsListener
+        } else {
+            sharedFilteredMappingMissionsFilter = filter
+            sharedFilteredMappingMissionsQuery = missionsQuery
+            sharedFilteredMappingMissionsListener = missionsListener
+        }
+    }
+
+    override fun updatePrivateFilteredMappingMissions(ownerUid: String, filter: String?) {
+        if (filter != null && privateFilteredMappingMissionsFilter != filter.toUpperCase()) {
```

Comment:
Cirrus raised a warning here with toUpperCase().
This hould maybe be
```suggestion
        if (filter != null && privateFilteredMappingMissionsFilter != filter.toUpperCase(Locale.ROOT)) {
```

Reply:
Done